packages/nextjs/components/ScaffoldEthAppWithProviders.tsx

```diff
@@ -4,29 +4,17 @@ import { useEffect, useState } from "react";
 import { RainbowKitProvider, darkTheme, lightTheme } from "@rainbow-me/rainbowkit";
 import { useTheme } from "next-themes";
 import { Toaster } from "react-hot-toast";
-import { WagmiConfig, useAccount } from "wagmi";
+import { SWRConfig } from "swr";
+import { WagmiConfig } from "wagmi";
 import { Footer } from "~~/components/Footer";
 import { Header } from "~~/components/Header";
 import { BlockieAvatar } from "~~/components/scaffold-eth";
 import { ProgressBar } from "~~/components/scaffold-eth/ProgressBar";
-import { useBGBuilderData } from "~~/hooks/useBGBuilderData";
-import { useGlobalState } from "~~/services/store/store";
 import { wagmiConfig } from "~~/services/web3/wagmiConfig";
 import { appChains } from "~~/services/web3/wagmiConnectors";
+import { fetcher } from "~~/utils/swr";
 
 const ScaffoldEthApp = ({ children }: { children: React.ReactNode }) => {
-  const { address } = useAccount();
-  const { data } = useBGBuilderData(address);
-
-  const setBuilderData = useGlobalState(state => state.setBuilderData);
-
-  useEffect(() => {
-    if (data?.id) {
-      setBuilderData(data);
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [setBuilderData, data?.id]);
```

Comment:
We won't be needing global state since in `useBGBuilderData` we have used [`useSWRImmutable`](https://swr.vercel.app/docs/revalidation#disable-automatic-revalidations) which will not make any request again for connected wallet and will give value from swr store whenever `useBGBuilderData` is called

Reply:
This is great